dashboard/src/components/AppView/ServicesTable/ServiceItem.tsx

```diff
@@ -1,32 +1,67 @@
 import * as React from "react";
+import { AlertTriangle } from "react-feather";
 
-import { IResource, IServiceSpec, IServiceStatus } from "../../../shared/types";
+import LoadingWrapper from "../../../components/LoadingWrapper";
+import { IKubeItem, IResource, IServiceSpec, IServiceStatus } from "../../../shared/types";
 
 interface IServiceItemProps {
-  service: IResource;
+  name: string;
+  service?: IKubeItem<IResource>;
+  getService: () => void;
 }
 
 class ServiceItem extends React.Component<IServiceItemProps> {
+  public componentDidMount() {
+    this.props.getService();
+  }
+
   public render() {
-    const { service } = this.props;
-    const spec: IServiceSpec = service.spec;
+    const { name, service } = this.props;
     return (
       <tr>
-        <td>{service.metadata.name}</td>
-        <td>{spec.type}</td>
-        <td>{spec.clusterIP}</td>
-        <td>{this.getExternalIP()}</td>
-        <td>
-          {spec.ports
-            .map(p => `${p.port}${p.nodePort ? `:${p.nodePort}` : ""}/${p.protocol || "TCP"}`)
-            .join(", ")}
-        </td>
+        <td>{name}</td>
+        {this.renderServiceInfo(service)}
       </tr>
     );
   }
 
-  private getExternalIP(): string {
-    const { service } = this.props;
+  private renderServiceInfo(service?: IKubeItem<IResource>) {
+    if (service === undefined || service.isFetching) {
+      return (
+        <td colSpan={4}>
+          <LoadingWrapper size="inline-small" />
```

Comment:
So this is going to be showing a loading message per element being loaded? It might be too much if we extrapolate this to let's say deployments no?

Do we need to show loading at all here? I thought that we already had some kind of struct that indicated that at least one resource was being loaded and we used that in the whole table?

Reply:
Correct, this was intended, we now show the name of the resource and a loading icon next to it. Each resource is loaded independently in parallel so it's possible that some are still loading whilst others have data to show.

> It might be too much if we extrapolate this to let's say deployments no?

What will be too much? In most cases it loads pretty fast and I don't think the spinners are too annoying (that said I haven't seen what this looks like on a chart with a lot of Services).

> Do we need to show loading at all here? I thought that we already had some kind of struct that indicated that at least one resource was being loaded and we used that in the whole table?

The table used to ensure all Services were loaded before displaying anything, but now it just renders `ServiceItem` containers. We could wrap the ServiceTable into a container and have a single loader, but I actually _like_ that we show each Service loading independently, it feels more responsive to me.

Reply:
For me it's okay to have a loading message per element. If having so many spinners is annoying we can remove it and have just a "Loading..." message.

Reply:
This is what it looks like, I quite like it. IMO it's subtle and not that annoying.

![kubeapps-serviceitemloading](https://user-images.githubusercontent.com/1544861/51286032-1c34d200-19a6-11e9-9a58-9c0321427392.gif)

Reply:
It does not look bad actually. I still have a preference for a global (per table) loading spinner but I understand that it complicates and couples the code, so I'd not push for it.

Just being picky and since asking for things is free. Something that annoys me is the alignment of the spinners. Something very slick would be the use of blurry lines a.l.a Facebook or Slack https://kyusuf.com/post/fake-it-til-you-make-it-css/